KeywordSearch/src/org/sleuthkit/autopsy/keywordsearch/KeywordSearchIngestModule.java

```diff
@@ -212,15 +214,43 @@ public ProcessResult process(AbstractFile abstractFile) {
             //skip indexing of virtual dirs (no content, no real name) - will index children files
             return ProcessResult.OK;
         }
+        
+        String detectedFormat = null;
+        ImageExtractor.SupportedFormats requestedFormat = null;
 
         if (KeywordSearchSettings.getSkipKnown() && abstractFile.getKnown().equals(FileKnown.KNOWN)) {
             //index meta-data only
-            indexer.indexFile(abstractFile, false);
+            detectedFormat = indexer.indexFile(abstractFile, false);
+            
+            // Check if the detected non-null format is supported. If supported,
+            // perform image extraction.
+            if (detectedFormat != null) {
+                for (ImageExtractor.SupportedFormats s : ImageExtractor.SupportedFormats.values()) {
+                    if (s.toString().equals(detectedFormat)) {
+                        requestedFormat = s;
+                    }
+                }
+                if (requestedFormat != null) {
+                    imageExtractor.extractImage(requestedFormat, abstractFile);
+                }
+            }
+            
+            
             return ProcessResult.OK;
         }
 
         //index the file and content (if the content is supported)
-        indexer.indexFile(abstractFile, true);
+        detectedFormat = indexer.indexFile(abstractFile, true);
```

Comment:
If indexFile is not able to detect the format, image extraction is not performed.